### PR TITLE
Revert "fix races in in-memory cache system (#18243)"

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

This reverts commit ebb6b47a9b03bfed439772465508696ccb0761f2.
This change had unintended side effects.
Reverting the commit may bring back assertion failures in the in-memory cache in maintainer mode during Jenkins test runs.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [x] Backport for 3.10: https://github.com/arangodb/arangodb/pull/18477
  - [x] Backport for 3.9: https://github.com/arangodb/arangodb/pull/18476
  - [ ] Backport for 3.8: not affected

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Cache/Cache.cpp [hunk 1]
```diff
@@ -343,10 +343,7 @@ void Cache::shutdown() {
       }
 
       SpinUnlocker taskUnguard(SpinUnlocker::Mode::Write, _taskLock);
-      SpinUnlocker shutdownUnguard(SpinUnlocker::Mode::Write, _shutdownLock);
-
-      // sleep a bit without holding the locks
-      std::this_thread::sleep_for(std::chrono::microseconds(20));
```
Comment (MBkkt) [inline arangod/Cache/Cache.cpp:349]: Hmm. why did you change it in reverted commit?
Because additional lock?

**Review by MBkkt — APPROVED** [review]

File: arangod/Cache/Cache.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/Cache/Cache.cpp:349]: I reverted the entire previous commit.
The version before had known issues, but these were probably less severe than the issues introduced by the commit that is now reverted.

Comment (MBkkt) [reply arangod/Cache/Cache.cpp:349]: I understand but just curios why in reverted version you decide to increase sleep_for time
